signature/policy_config.go

```diff
@@ -174,7 +172,7 @@ func (m *policyTransportScopesWithTransport) UnmarshalJSON(data []byte) error {
 		if _, ok := tmpMap[key]; ok {
 			return nil
 		}
-		if key != "" {
+		if key != "" && m.transport != nil {
```

Comment:
That tests parsing the policy into a data structure, not using the data structure to look up the relevant rules to evaluate for a particular image (`requirementsForImageRef`). Right now, by inspection, it seems clear enough that `requirementsForImageRef` can correctly handle a transport which exists but is unregistered, or a transport which was not registered when parsing the policy but is registered afterwards—but the transport registry is a very non-obvious hidden state in the parsing/evaluation interaction, so it would be nice to have a test which ensures that the code keeps working in these corner cases.